From fbc61efe6e80a5af2e6e3f7a8b0813cfe66e34de Mon Sep 17 00:00:00 2001
From: Nadia Santalla <nadia@santalla.io>
Date: Tue, 8 Jul 2025 17:54:12 +0200
Subject: [PATCH] owocafe: add warning when posting in public

---
 .../features/compose/components/warning.tsx     | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/app/javascript/mastodon/features/compose/components/warning.tsx b/app/javascript/mastodon/features/compose/components/warning.tsx
index f862f84cba..3bbe8a6071 100644
--- a/app/javascript/mastodon/features/compose/components/warning.tsx
+++ b/app/javascript/mastodon/features/compose/components/warning.tsx
@@ -17,11 +17,12 @@ const selector = createSelector(
     needsLockWarning: privacy === 'private' && !locked,
     hashtagWarning: privacy !== 'public' && HASHTAG_PATTERN_REGEX.test(text),
     directMessageWarning: privacy === 'direct',
+    owocafePublicWarning: privacy === 'public' && text !== "",
   }),
 );
 
 export const Warning = () => {
-  const { needsLockWarning, hashtagWarning, directMessageWarning } =
+  const { needsLockWarning, hashtagWarning, directMessageWarning, owocafePublicWarning } =
     useAppSelector(selector);
   if (needsLockWarning) {
     return (
@@ -72,6 +73,20 @@ export const Warning = () => {
     );
   }
 
+  if (owocafePublicWarning) {
+    return (
+      <WarningMessage>
+        <FormattedMessage
+          id='compose_form.owocafe_public_warning'
+          defaultMessage="Este post se mostrará a todos los usuarios de owo.cafe. Tenemos {rules} respecto a la temática y tono de los con visibilidad 🌍 Pública."
+          values={{
+            rules: <a href="https://wiki.owo.cafe/owocafe/timeline-publica" target="_blank">normas específicas</a>
+          }}
+        />
+      </WarningMessage>
+    );
+  }
+
   return null;
 };
 
-- 
2.50.1

