From d7edacb2e539f3b56509718ae9e3911526751c63 Mon Sep 17 00:00:00 2001
From: Nadia Santalla <nadia@nadia.moe>
Date: Wed, 22 Jul 2026 16:02:26 +0200
Subject: [PATCH] owocafe: add warning when posting in public

---
 .../features/compose/components/warning.tsx     | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/app/javascript/mastodon/features/compose/components/warning.tsx b/app/javascript/mastodon/features/compose/components/warning.tsx
index 875d6df58d..b5d12a832d 100644
--- a/app/javascript/mastodon/features/compose/components/warning.tsx
+++ b/app/javascript/mastodon/features/compose/components/warning.tsx
@@ -17,11 +17,12 @@ const selector = createSelector(
     needsLockWarning: privacy === 'private' && !locked,
     hashtagWarning: privacy !== 'public' && HASHTAG_PATTERN_REGEX.test(text),
     directMessageWarning: privacy === 'direct',
+    owocafePublicWarning: privacy === 'public' && text !== "",
   }),
 );
 
 export const Warning = () => {
-  const { needsLockWarning, hashtagWarning, directMessageWarning } =
+  const { needsLockWarning, hashtagWarning, directMessageWarning, owocafePublicWarning } =
     useAppSelector(selector);
   if (needsLockWarning) {
     return (
@@ -76,6 +77,20 @@ export const Warning = () => {
     );
   }
 
+  if (owocafePublicWarning) {
+    return (
+      <WarningMessage>
+        <FormattedMessage
+          id='compose_form.owocafe_public_warning'
+          defaultMessage="Este post se mostrará a todos los usuarios de owo.cafe. Tenemos {rules} respecto a la temática y tono de los con visibilidad 🌍 Pública."
+          values={{
+            rules: <a href="https://wiki.owo.cafe/owocafe/timeline-publica" target="_blank">normas específicas</a>
+          }}
+        />
+      </WarningMessage>
+    );
+  }
+
   return null;
 };
 
-- 
2.55.0

