From 8e3add3e101fb16c0fe627536f3965342bcdcc90 Mon Sep 17 00:00:00 2001
From: Nadia Santalla <nadia@santalla.io>
Date: Tue, 8 Jul 2025 18:59:41 +0200
Subject: [PATCH] owocafe: add link to wiki

---
 .../mastodon/features/navigation_panel/index.tsx         | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/app/javascript/mastodon/features/navigation_panel/index.tsx b/app/javascript/mastodon/features/navigation_panel/index.tsx
index b62e8f9fa6..155e25b586 100644
--- a/app/javascript/mastodon/features/navigation_panel/index.tsx
+++ b/app/javascript/mastodon/features/navigation_panel/index.tsx
@@ -26,6 +26,7 @@ import SettingsIcon from '@/material-icons/400-24px/settings.svg?react';
 import StarActiveIcon from '@/material-icons/400-24px/star-fill.svg?react';
 import StarIcon from '@/material-icons/400-24px/star.svg?react';
 import TrendingUpIcon from '@/material-icons/400-24px/trending_up.svg?react';
+import WikiIcon from '@/material-icons/400-24px/edit_note-fill.svg?react';
 import { fetchFollowRequests } from 'mastodon/actions/accounts';
 import { openNavigation, closeNavigation } from 'mastodon/actions/navigation';
 import { Account } from 'mastodon/components/account';
@@ -224,6 +225,14 @@ export const NavigationPanel: React.FC<{ multiColumn?: boolean }> = ({
       {banner && <div className='navigation-panel__banner'>{banner}</div>}
 
       <div className='navigation-panel__menu'>
+        <ColumnLink
+          to='https://wiki.owo.cafe'
+          icon='wiki'
+          iconComponent={WikiIcon}
+          activeIconComponent={WikiIcon}
+          text='Wiki'
+        />
+
         {signedIn && (
           <>
             {!multiColumn && (
-- 
2.50.0

